### PR TITLE
feat: add deprecation rfc for this rfcs repo

### PR DESCRIPTION
**Pull request by DSchau**
First of all -- how very _meta_ 😄 

This `rfcs` repository has unfortunately not been given the care and attention that such crucial and sweeping changes deserve. One seemingly obvious reason is that the `rfcs` repo is separate from the GatsbyJS monorepo which gets much more traffic and attention from not only the community but also the GatsbyJS open-source team (@gatsbyjs/core).

From this perspective, I propose a revamp of this underlying model to more effectively surface this valuable community feedback and ensure that the community has ample opportunity to help shape and influence Gatsby open-source, with a direct line of feedback to the open-source team. Specifically, I propose:

1. We deprecate this repo and close any outstanding pull requests in this repo
1. We document and encourage users to open up RFCs as pull requests to the official GatsbyJS monorepo (`gatsbyjs/gatsby`)

## 📝 [View Rendered Text](https://github.com/DSchau/rfcs/blob/rfcs/move-rfcs-to-monorepo/text/0000-move-rfcs-to-monorepo.md)

### CODE REVIEW

**Comment by DSchau** [comment]
@sidharthachatterjee great questions! Lemme answer 'em 👍 

> I'm not sure if we should block on merging a PR for an RFC on the implementation of said RFC itself since that would prohibit visibility. Perhaps when merging an RFC, we open an issue to track the implementation of said RFC (with takeaways from it)?

I left it open a bit -- I generally agree. This being said, I think if we think of the RFC as the Gatsby open-source design document (which it is) we need some mechanism/trigger to close. The PR feels about as natural as they come.

So perhaps it may not be the implementor of the RFC that closes it (which can be merged earlier) but perhaps the PR that implements the RFC changes the status of the RFC, e.g.

```diff
- status: MERGED
- status: PUBLISHED
```

or something :) Happy for feedback here!

> One question: Do we want to move over existing open RFCs in this repo?

I vote no. I think the "forcing function" of moving over the RFCs will separate the ones worth doing from the ones that may not be relevant anymore. If the authors move them over -- that's great! If not, perhaps that's an OK side effect. If there are some that are worth doing that the community/author hasn't moved over -- the core team can manually do so!

**Comment by blainekasten** [comment]
<3 this!

Are we going to have a RFC issue label? I feel that would help to filter and be able to quickly find rfcs in progress

> I vote no. I think the "forcing function" of moving over the RFCs will separate the ones worth doing from the ones that may not be relevant anymore. If the authors move them over -- that's great! If not, perhaps that's an OK side effect. If there are some that are worth doing that the community/author hasn't moved over -- the core team can manually do so!

I think we should drop a note on each one to let the authors know they should move it over if they want it to continue to exist.

**Comment by DSchau** [comment]
> Are we going to have a RFC issue label? I feel that would help to filter and be able to quickly find rfcs in progress

We definitely should. I didn't spell it out (but should have!) but I'd propose `type: rfc` or something!

> I think we should drop a note on each one to let the authors know they should move it over if they want it to continue to exist.

Agreed! I'll add that to the doc.

File: text/0000-move-rfcs-to-monorepo.md [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+
+I contend that they aren't really separate concerns. A successful RFC leads to a merged pull request _in_ the open-source repo, so from this perspective it's reasonable to conclude that opening up an RFC alongside the eventual pull request makes intuitive sense and is worth doing.
+
+# Alternatives
+
+There are a few alternatives worth considering. Specifically:
+
+## Keep this repo as-is
+
+We could not do this (of course) and keep the status quo. This has no drawbacks, and one positive in their is zero effort required to keep the status quo.
```
Comment (DSchau) [inline text/0000-move-rfcs-to-monorepo.md]:
```suggestion
We could not do this (of course) and keep the status quo. This has no drawbacks, and one positive in that there is zero effort required to keep the status quo.
```

**Comment by DSchau** [comment]
Looks like this is good to go :) I'll write a little blog post, and then merge this later this week.

Thanks all!

**Comment by wardpeet** [comment]
ping @DSchau

**Comment by DSchau** [comment]
Later this week is up for interpretation :sweat-smile:

I half wrote the blog post, but it stalled a bit. I'll re-visit later this week and get this merged. I'd like to couple the blog post with the introduction of a "new" process so we can maximize it's efficacy.

**Comment by DSchau** [comment]
Alright -- let's merge this. We'll "soft deprecate" this repo for now, and then we'll archive once we create some documentation and the coupling blog post.

**Pull request merged**